python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1303 @@
+            detector = dataRef.get('camera')[dataRef.dataId[self.config.ccdKey]]
+            ampInfoCat = detector.getAmpInfoCatalog()
+            ampNames = [amp.getName() for amp in ampInfoCat]
+            xcorrs = {key: [] for key in ampNames}
+            means = {key: [] for key in ampNames}
+        else:
+            raise RuntimeError("Unsupported level: {}".format(self.config.level))
+
+        # calculate or retrieve the gains
+        if self.config.doCalcGains:
+            self.log.info('Beginning gain estimation for detector %s' % detNum)
+            gains, nomGains = self.estimateGains(dataRef, visitPairs)
+            dataRef.put(gains, datasetType='brighterFatterGain')
+            self.log.info('Finished gain estimation for detector %s' % detNum)
+        else:
+            gains = dataRef.get('brighterFatterGain')
+            if not gains:
+                raise RuntimeError('Failed to retrieved gains for detector %s' % detNum)
+            self.log.info('Retrieved stored gain for detector %s' % detNum)
+        self.log.debug('Detector %s has gains %s' % (detNum, gains))
+
+        # Loop over pairs of visits, calculating the cross-correlations at the required level
+        for (v1, v2) in visitPairs:
+
+            dataRef.dataId['visit'] = v1
+            exp1 = self.isr.runDataRef(dataRef).exposure
+            dataRef.dataId['visit'] = v2
+            exp2 = self.isr.runDataRef(dataRef).exposure
+            del dataRef.dataId['visit']
+
+            self.log.info('Preparing images for cross-correlation calculation for detector %s' % detNum)
+            # note the shape of these returns depends on level
+            _scaledMaskedIm1, _means1 = self._prepareImage(exp1, gains, self.config.level)
+            _scaledMaskedIm2, _means2 = self._prepareImage(exp2, gains, self.config.level)
+
+            if self.debug.enabled:
+                try:
+                    frameId1 = exp1.getMetadata().get("FRAMEID")
+                    frameId2 = exp2.getMetadata().get("FRAMEID")
+                    frameId = '_diff_'.join(frameId1, frameId2)
+                except:
+                    frameId = 'Im1 diff Im2'
+
+            # Compute the cross-correlation and means at the appropriate config.level:
+            # - "DETECTOR": one key, so compare the two visits to each other
+            # - "AMP": n_amp keys, comparing each amplifier of one visit to the same amplifier in other visit
+            for det_object in _scaledMaskedIm1.keys():
+                xcorrs[det_object].append(self._crossCorrelate(_scaledMaskedIm1[det_object],
+                                                               _scaledMaskedIm2[det_object]))
+                means[det_object].append([_means1[det_object], _means2[det_object]])
+
+            # this is awkward now that code is refactored
+            # not sure how best to make this work without passing everything around
```

Comment:
I'm not seeing the issue; can you please explain more (here in github is fine, no need to modify the code)

Reply:
The solution here is going to be pull out all the commented code and ticket enhancing the debug plot functionality I think. But to explain though:

the issue is that to generate the correct titles for the debug plots you need the objects, but you don't want to be passing everything around just to be able to query things that aren't really used. However, a massive block of code to generate the titles doesn't really belong in the main routine, and even less so as it mostly won't even fire.

I don't think there's a nice solution to this, and we'll probably see this code reappear in this place, though hopefully a little more cleanly/concisely, if/when that ticket gets done, but for now I think just removing it is fine.